From acfc4e7e1b662a3b607de5e14269fce0bfeb6f68 Mon Sep 17 00:00:00 2001
From: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Date: Thu, 23 Mar 2023 17:30:30 +0200
Subject: [PATCH 2/3] drivers: s32cc_cmu: update expected qspi frequency

The QSPI_1X frequency is 133 MHz for targets on EVB Boards and 200 MHz
for the rest.

Issue: ALB-9840
Upstream-Status: Pending

Signed-off-by: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/misc/s32cc_cmu.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/drivers/misc/s32cc_cmu.c b/drivers/misc/s32cc_cmu.c
index ee2687dc45..a726be6679 100644
--- a/drivers/misc/s32cc_cmu.c
+++ b/drivers/misc/s32cc_cmu.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Copyright 2021-2022 NXP
+ * Copyright 2021-2023 NXP
  */
 
 #include <common.h>
@@ -45,6 +45,13 @@
 
 #define MAX_DEPTH 20
 
+#if defined(CONFIG_TARGET_S32G2XXAEVB) || defined(CONFIG_TARGET_S32R45EVB) || \
+defined(CONFIG_TARGET_S32G3XXAEVB)
+#define QSPI_1X_MAX_FREQ 133333333
+#else
+#define QSPI_1X_MAX_FREQ 200000000
+#endif
+
 struct s32cc_cmu_data {
 	struct cmu *cmu_blocks;
 	int n_blocks;
@@ -111,7 +118,7 @@ static struct cmu s32g2_cmu_blocks[] = {
 	FXOSC_PERIPH_CMU_FC_RANGE(14, GMAC_0_TX, 2500000, 125000000),
 	FXOSC_PERIPH_CMU_FC_RANGE(15, GMAC_TS, 5000000, 200000000),
 	FXOSC_PERIPH_CMU_FC(16, LIN, 125000000),
-	FXOSC_PERIPH_CMU_FC(17, QSPI_1X, 200000000),
+	FXOSC_PERIPH_CMU_FC_RANGE(17, QSPI_1X, 66666666, QSPI_1X_MAX_FREQ),
 	FXOSC_PERIPH_CMU_FC(18, SDHC, 400000000),
 
 	FIRC_PERIPH_CMU_FC(20, DDR, 666666666),
@@ -157,7 +164,7 @@ static struct cmu s32g3_cmu_blocks[] = {
 	FXOSC_PERIPH_CMU_FC_RANGE(14, GMAC_0_TX, 2500000, 125000000),
 	FXOSC_PERIPH_CMU_FC_RANGE(15, GMAC_TS, 5000000, 200000000),
 	FXOSC_PERIPH_CMU_FC(16, LIN, 125000000),
-	FXOSC_PERIPH_CMU_FC(17, QSPI_1X, 200000000),
+	FXOSC_PERIPH_CMU_FC_RANGE(17, QSPI_1X, 66666666, QSPI_1X_MAX_FREQ),
 	FXOSC_PERIPH_CMU_FC(18, SDHC, 400000000),
 
 	FIRC_PERIPH_CMU_FC(20, DDR, 800000000),
@@ -201,7 +208,7 @@ static struct cmu s32r45_cmu_blocks[] = {
 	FXOSC_PERIPH_CMU_FC(14, GMAC_0_TX, 125000000),
 	FXOSC_PERIPH_CMU_FC(15, GMAC_TS, 200000000),
 	FXOSC_PERIPH_CMU_FC(16, LIN, 125000000),
-	FXOSC_PERIPH_CMU_FC(17, QSPI_1X, 133333333),
+	FXOSC_PERIPH_CMU_FC_RANGE(17, QSPI_1X, 66666666, QSPI_1X_MAX_FREQ),
 	FXOSC_PERIPH_CMU_FC(18, SDHC, 400000000),
 
 	FIRC_PERIPH_CMU_FC(20, DDR, 800000000),
-- 
2.25.1

